python-checks/src/main/java/org/sonar/python/checks/regex/ReluctantQuantifierCheck.java

```diff
@@ -21,15 +21,29 @@
 
 import org.sonar.check.Rule;
 import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonarsource.analyzer.commons.regex.RegexIssueReporter;
 import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.ast.Quantifier;
 import org.sonarsource.analyzer.commons.regex.finders.ReluctantQuantifierFinder;
 
 @Rule(key = "S5857")
 public class ReluctantQuantifierCheck extends AbstractRegexCheck {
 
   @Override
   public void checkRegex(RegexParseResult regexParseResult, CallExpression regexFunctionCall) {
-    new ReluctantQuantifierFinder(this::addIssue).visit(regexParseResult);
+    new PythonReluctantQuantifierFinder(this::addIssue).visit(regexParseResult);
+  }
+
+  private static class PythonReluctantQuantifierFinder extends ReluctantQuantifierFinder {
+
+    public PythonReluctantQuantifierFinder(RegexIssueReporter.ElementIssue regexElementIssueReporter) {
+      super(regexElementIssueReporter);
+    }
+
+    @Override
+    protected String makePossessiveOrGreedy(Quantifier quantifier, boolean possessive) {
+      return super.makePossessiveOrGreedy(quantifier, false);
```

Comment:
One alternative could be to check whether the Python version set for the analyzer is >= 3.11 (in which case we could actually make the quantifier possessive).
No need to do it now, but maybe we could have a comment for that.